### ISSUE TITLE
Fit to Window Size

### ISSUE THREAD

**Issue by metalx1000** [Awesome-Game-Team/Awesome-Game#27]
I've been working on trying to get the game to fit to the size of the browser window.
I think it would be nice to have that on startup and also I know iOS devices (last I checked) are unable to go into full screen mode.  I've created a "resizeGame" function, that almost works. Everything resizes, except the map layer doesn't seem to render out to the full screen.  See what I mean here:
http://filmsbykris.com/scripts/Awesome-Game/?fitScreen=ture

The function is in the "input.js" file right now (although I think we should make a "display.js" and put this function and fullscreen functions in there).  If anyone can figure out what I'm missing I would appreciate it.  I'm thinking it might be a camera resizeing issue, but I played with that and didn't get anywhere.

**Comment by jonpatterns** [comment]
Setting the renderer and camera size is needed.  However, it looks like each layer of the tilemap needs to be resized too, for example:

If you store tilemap handles using variables `layerbackdrop` and `layer` when they are created in map.js:

```
layerbackdrop = map.createLayer('backdrop');
layer = map.createLayer('world');
```

Then this resize function works in input.js:

```
  function resizeGame() {
  var height = window.innerHeight;
  var width = window.innerWidth;
  game.width = width;
  game.height = height;
  game.renderer.resize(width, height);
  game.camera.setSize(width, height);
  layer.resize(width, height);
  layerbackdrop.resize(width, height);
}
```

I've got a suspicion Phaser can handle resizing across different devices using Phaser.ScaleManager or other inbuilt functions.

**Comment by OneSleepyDev** [comment]
I'm having some success with this in input.js, except the icon and text display are off some. (Which might be due to the bounds line)
I agree to the display.js file

```
var width = window.innerWidth * window.devicePixelRatio / 1.5;
var height = window.innerHeight * window.devicePixelRatio /1.5;
Phaser.ScaleManager.SHOW_ALL = true;
game.camera.setSize(width, height);
game.camera.setBoundsToWorld();
game.renderer.resize(width, height);
```

Is that close to what you're looking for?

**Comment by metalx1000** [comment]
@jonpatterns Resizing the the layers worked, but there are two issues now.
1) when you go into fullscreen, there is now letteboxing happening (not a big deal).
2)when coming out of fullscreen mode hud and messages are don't get re-positioned.
This can be fixed by resetting there "cameraOffset.setTo".  

I'll need to put each button in an array when created and set default positioning for them.
This should be pretty easy, and probably should done anyway.
But, I'm tired tonight.  I'll try and work on it tomorrow.

**Comment by jonpatterns** [comment]
The resize routine can be omitted by setting 'game.scale.fullScreenScaleMode' and 'game.scale.scaleMode' when the game initialises. For example,

```
game.scale.fullScreenScaleMode = Phaser.ScaleManager.SHOW_ALL; 
game.scale.scaleMode = Phaser.ScaleManager.SHOW_ALL;
```

The SHOW_ALL option even dynamically resizes when the browser windows size is readjusted. 
http://phaser.io/docs/2.3.0/Phaser.ScaleManager.html

Downsides:
- the game is slightly letterboxed - there may be a ratio/camera attribute to change this.
- you don't get an expanded view of a level, just zoomed in. There maybe other 'scale modes' that allow this if desired.
- minor bug - using escape key to exit fullscreen does not update fullscreen toggle button.

**Comment by metalx1000** [comment]
I think I've got it mostly working with @jonpatterns suggestions leading me in the right directions.
I got it fullpage, no letterboxing on fullscreen, and it resizes correctly when you come out of fullscreen mode.

The problem at this point is if the game started in Portrait mode on a mobile device it screws everything up.  So we need a screen that tells the player to turn their device landscape, if it's not already, before the game will load.

I'm going to start a new "issue" for that.
